src/rules/sortImportsRule.ts

```diff
@@ -0,0 +1,244 @@
+
+    // See if the import type is still available
+    const index = this.expectedOrder.indexOf(importData.memberSyntaxType, this.currentImportIndex);
+    if (index !== -1) {
+      if (this.expectedOrder[this.currentImportIndex] !== importData.memberSyntaxType) {
+        this.currentImportIndex = index;
+        this.currentSortValue = {
+          sortValue: this.caseConverter(importData.sortValue),
+          originalValue: importData.sortValue
+        };
+      } else if (this.currentSortValue.sortValue > this.caseConverter(importData.sortValue)) {
+        this.addFailureAtNode(
+          node,
+          `All imports of the same type must be sorted alphabetically. "${importData.sortValue}" must come before "${this.currentSortValue.originalValue}"`);
+      } else {
+        this.currentSortValue = {
+          sortValue: this.caseConverter(importData.sortValue),
+          originalValue: importData.sortValue
+        };
+      }
+    } else {
+      const currentSyntaxType = MemberSyntaxType[importData.memberSyntaxType];
+      const previousSyntaxType = MemberSyntaxType[this.expectedOrder[this.currentImportIndex]];
+      this.addFailureAtNode(
+        node,
+        `All imports of type "${currentSyntaxType}" must occur before all imports of type "${previousSyntaxType}"`);
+    }
+  }
+
+  private _determineImportType(node: ts.ImportDeclaration | ts.ImportEqualsDeclaration): ImportMetadata {
+    const nodeText = node.getFullText();
+
+    if (node.kind === ts.SyntaxKind.ImportEqualsDeclaration) {
+      const aliasMatch = /\bimport\s+(\w+)\s*=.+/g.exec(nodeText);
+      return {
+        memberSyntaxType: MemberSyntaxType.Alias,
+        sortValue: aliasMatch[1]
+      };
+    } else {
+      const singleMatch = /\bimport\s+({?([^,{}\*]+?)}?)\s*from\s+[\'"](?:[^"\']+)["\']/g.exec(nodeText);
+      const multipleMatch = /\bimport\s*{?\s*([^{}\'",]+?)\s*,(?:\s*.+\s*,\s*)*\s*.+\s*}?\s*from\s+[\'"](?:[^"\']+)["\']/g.exec(nodeText);
+      const noneMatch = /\bimport\s+[\'"]([^"\']+)["\']/g.exec(nodeText);
+      const allMatch = /\bimport\s+\*\s+as\s+(.+)\s+from\s+[\'"](?:[^"\']+)["\']/g.exec(nodeText);
+
+      let result;
+      if (singleMatch !== null) {
+        result = {
+          memberSyntaxType: MemberSyntaxType.Single,
+          sortValue: singleMatch[1]
+        };
+      } else if (multipleMatch !== null) {
+        result = {
+          memberSyntaxType: MemberSyntaxType.Multiple,
+          sortValue: multipleMatch[1]
+        };
+      } else if (noneMatch !== null) {
+        result = {
+          memberSyntaxType: MemberSyntaxType.None,
+          sortValue: noneMatch[1]
+        };
+      } else if (allMatch !== null) {
+        result = {
+          memberSyntaxType: MemberSyntaxType.All,
+          sortValue: allMatch[1]
+        };
+      }
+      else {
+        this.addFailureAtNode(node, 'Could not determine import type');
+      }
+      return result;
```

Comment:
That was going to be another suggestion but I only thought about it after the initial review. This is much clearer :).